exercises/bowling/example.py

```diff
@@ -0,0 +1,157 @@
+
+            else:
+
+                if(self.isSpare(roll1, roll2)):
+                    self.totalScore += roll1 + roll2 + \
+                                       self.spareBonus(frame_index)
+                else:
+                    self.totalScore += roll1 + roll2
+
+            frame_index += 1
+
+        return self.totalScore
+
+    def isStrike(self, pins):
+        if (pins == MAX_PINS):
+            return True
+        else:
+            return False
+
+    def isSpare(self, roll1, roll2):
+        if (roll1 + roll2 == MAX_PINS):
+            return True
+        else:
+            return False
+
+    def stikeBonus(self, frame_index):
+        bonusroll1 = self.rolls[frame_index+1].getFrame()[0]
+        bonusroll2 = 0
+        # need to go further out if the next on is a strike
+        if (bonusroll1 == 10):
+            bonusroll2 = self.rolls[frame_index+2].getFrame()[0]
+        else:
+            bonusroll2 = self.rolls[frame_index+1].getFrame()[1]
+        # edge case - if the last roll is a stike the bonus rolls needs to be
+        # validated
+        if (not self.isStrike(bonusroll1) and
+           (bonusroll1 + bonusroll2 > MAX_PINS)):
+            raise ValueError
+        else:
+            return bonusroll1 + bonusroll2
+
+    def spareBonus(self, frame_index):
+        return self.rolls[frame_index+1].getFrame()[0]
+
+    def isLastFrame(self, frame_index):
+        if(frame_index >= len(self.rolls)-1):
+            return True
+        return False
+
+    def bonusRollsEarned(self):
+        if (len(self.rolls) == NUM_FRAMES):
+            lastFrame = self.rolls[NUM_FRAMES-1].getFrame()
+            if (self.isStrike(lastFrame[0])):
+                self.bonusRollsAccrued = 2
+            elif (self.isSpare(lastFrame[0], lastFrame[1])):
+                self.bonusRollsAccrued = 1
+        else:
+            self.bonusRollsAccrued = 0
+        return
+
+    def isBonusRoll(self):
+        # if we've already seen all
+        if (len(self.rolls) >= NUM_FRAMES):
+            return True
+        else:
+            return False
+
+
+class Frame(object):
+
+    def __init__(self):
+        self.rolls = [None, None]
+        self.open = True
+
+    def roll(self, roll, bonusRoll, accruedBonuses, seenBonuses):
+        # if it's a strike we close the frame
+        if (roll == 10):
+            self.rolls[0] = 10
+            self.rolls[1] = 0
+            self.open = False
+        else:
+            # first roll, but frame is still open
+            if (self.rolls[0] is None):
+                self.rolls[0] = roll
+                # may need to close bonus roll frames before 2 have been seen
+                if (bonusRoll and seenBonuses == accruedBonuses):
+                    self.rolls[1] = 0
+                    self.open = False
+            else:
+                # second roll, closes frame
+                self.rolls[1] = roll
+                self.open = False
+
+    def isOpen(self):
```

Comment:
this doesn't look right, method name doesn't correspond to its actual function

Reply:
I'm not sure what you mean on this one.  you call isOpen to find out if the frame is open (true) or not (false)